holonote/annotate/plotting.py

```diff
@@ -0,0 +1,505 @@
+from __future__ import annotations
+
+import weakref
+from typing import TYPE_CHECKING, Any
+
+import holoviews as hv
+import pandas as pd
+import param
+from bokeh.models.tools import BoxSelectTool, HoverTool, Tool
+
+from .._warnings import warn
+
+if TYPE_CHECKING:
+    from .annotator import Annotator
+
+
+class _StyleOpts(param.Dict):
+    def _validate(self, val) -> None:
+        super()._validate(val)
+        for k in val:
+            if k in ("color", "alpha"):
+                warn("Color and alpha opts should be set directly on the style object.")
+
+
+_default_opts = {"apply_ranges": False, "show_legend": False}
+
+
+class Style(param.Parameterized):
+    alpha = param.Number(default=0.2, bounds=(0, 1), allow_refs=True)
+    alpha.doc = "Alpha value for non-selected regions"
+
+    selection_alpha = param.Number(default=0.7, bounds=(0, 1), allow_refs=True)
+    selection_alpha.doc = "Alpha value for selected regions"
+
+    edit_alpha = param.Number(default=0.4, bounds=(0, 1), allow_refs=True)
+    edit_alpha.doc = "Alpha value for editing regions"
+
+    color = param.Parameter(default="red", doc="Color of the indicator", allow_refs=True)
+    edit_color = param.Parameter(default="blue", doc="Color of the editor", allow_refs=True)
+    selection_color = param.Parameter(
+        default=None, doc="Color of selection, by the default the same as color", allow_refs=True
+    )
+
+    # Indicator opts (default and selection)
+    opts = _StyleOpts(default={})
+    line_opts = _StyleOpts(default={})
+    span_opts = _StyleOpts(default={})
+    rectangle_opts = _StyleOpts(default={})
+
+    # Editor opts
+    edit_opts = _StyleOpts(default={"line_color": "black"})
+    edit_line_opts = _StyleOpts(default={})
+    edit_span_opts = _StyleOpts(default={})
+    edit_rectangle_opts = _StyleOpts(default={})
+
+    @property
+    def indicator_selection(self) -> dict[str, tuple]:
```

Comment:
I think `indicator` is implicitly always the case...so I would considered `style.selected` and `style.non_selected`.

Reply:
It is a little weird that the selected options are returned here as a dictionary while the non-selected options are returned in the `indicator` method as a tuple of `hv.opts` objects.

Might not be a code change but I think there should be a docstring to say that the options returned by this method are bokeh options that are expected to be supported by *all* indicator types.

Reply:
This expands the existing logic of `indicator_highlight` (you created) for later use in a `hv.dim`. The `hv.dim` is then passed into `indicator` as `**select_opts`. 

I will make the property private.

Reply:
I'm not sure making the property private is necessary....but what about the suggested name change?